integration/helper_test.go

```diff
@@ -433,6 +434,160 @@ options:
+
+			patternHelperPid := fmt.Sprintf(":%06d", helperCmd.Process.Pid)
+			helperOut, _ := exec.Command("grep", patternHelperPid, helperFiles[len(helperFiles)-1]).CombinedOutput()
+			helperOutput := string(helperOut)
+
+			// Batch 0 should be processed
+			Expect(helperOutput).To(ContainSubstring(`: Skip file discovered, skipping this relation`))
+			Expect(helperOutput).To(ContainSubstring(`Segment 1: Oid 1, Batch 0: Opening pipe`))
+
+			// Batch 2 must not be processed
+			Expect(helperOutput).ToNot(ContainSubstring(`Segment 1: Oid 1, Batch 2: Skip file discovered, skipping this relation`))
+			Expect(helperOutput).ToNot(ContainSubstring(`Segment 1: Oid 1, Batch 2: Opening pipe`))
+		})
+		It("skips batches if skip file is discovered with resize restore", func() {
+			// Run helper only with restore for a few batches and skip file defined
+			filesToDelete := setupRestoreWithSkipFiles(1, false)
+			for _, f := range filesToDelete {
+				defer func(filename string) {
+					os.Remove(filename)
+				}(f)
+			}
+
+			By("Create restore command")
+			args := []string{
+				"--toc-file", tocFile,
+				"--oid-file", restoreOidFile,
+				"--pipe-file", pipeFile,
+				"--content", "1",
+				"--resize-cluster",
+				"--orig-seg-count", "6",
+				"--dest-seg-count", "3",
+				"--restore-agent",
+				"--data-file", dataFileFullPath + ".gz",
+				"--on-error-continue",
+			}
+			helperCmd := exec.Command(gpbackupHelperPath, args...)
+
+			err := helperCmd.Start()
+			Expect(err).ToNot(HaveOccurred())
+
+			// Block here until gpbackup_helper finishes (cleaning up pipes)
+			err = helperCmd.Wait()
+			Expect(err).ToNot(HaveOccurred())
+			for _, i := range []int{1, 2, 3} {
+				currentPipe := fmt.Sprintf("%s_%d_0", pipeFile, i)
+				Expect(currentPipe).ToNot(BeAnExistingFile())
+			}
+
+			By("Check in logs that batches were not restored")
+
+			homeDir := os.Getenv("HOME")
+			helperFiles, _ := path.Glob(path.Join(homeDir, "gpAdminLogs/gpbackup_helper_*"))
+
+			Expect(helperFiles).NotTo(BeEmpty())
+
+			patternHelperPid := fmt.Sprintf(":%06d", helperCmd.Process.Pid)
+			helperOut, _ := exec.Command("grep", patternHelperPid, helperFiles[len(helperFiles)-1]).CombinedOutput()
+			helperOutput := string(helperOut)
+
+			fmt.Printf("pattern_helper_pid = %s", patternHelperPid)
+
+			Expect(helperOutput).ToNot(BeEmpty())
+
+			// Batch 0 should be processed
+			Expect(helperOutput).To(ContainSubstring(`: Skip file discovered, skipping this relation`))
+			Expect(helperOutput).To(ContainSubstring(`Segment 1: Oid 1, Batch 0: Opening pipe`))
+
+			// Batch 2 must not be processed
+			Expect(helperOutput).ToNot(ContainSubstring(`Segment 1: Oid 1, Batch 2: Skip file discovered, skipping this relation`))
+			Expect(helperOutput).ToNot(ContainSubstring(`Segment 1: Oid 1, Batch 2: Opening pipe`))
+		})
+		It("skips batches if skip file is discovered with resize restore using a plugin", func() {
```

Comment:
how about to add test
"skips batches if skip file is discovered with single datafile config using a plugin"?